lib/remote/apilistener.cpp

```diff
@@ -1050,82 +1061,88 @@ bool ApiListener::RelayMessageOne(const Zone::Ptr& targetZone, const MessageOrig
 
 	std::vector<Endpoint::Ptr> skippedEndpoints;
 
-	bool relayed = false, log_needed = false, log_done = false;
-
-	std::set<Endpoint::Ptr> targetEndpoints;
-
+	std::set<Zone::Ptr> allTargetZones;
 	if (targetZone->GetGlobal()) {
-		targetEndpoints = localZone->GetEndpoints();
-
+		/* if the zone is global, the message has to be relayed to our local zone and direct children */
+		allTargetZones.insert(localZone);
 		for (const Zone::Ptr& zone : ConfigType::GetObjectsByType<Zone>()) {
-			/* Fetch immediate child zone members */
 			if (zone->GetParent() == localZone) {
-				std::set<Endpoint::Ptr> endpoints = zone->GetEndpoints();
-				targetEndpoints.insert(endpoints.begin(), endpoints.end());
+				allTargetZones.insert(zone);
 			}
 		}
 	} else {
-		targetEndpoints = targetZone->GetEndpoints();
+		/* whereas if it's not global, the message is just relayed to the zone itself */
+		allTargetZones.insert(targetZone);
```

Comment:
And localZone?

Reply:
Why do you think it's needed? The old code also only considers targetZone in this case: https://github.com/Icinga/icinga2/blob/dc79c4e59a38971b148116d42e3c02372373620d/lib/remote/apilistener.cpp#L1068

I'm not 100% sure about this case as I mostly considered global zones so I kept it as it is.

Reply:
Fine.